cluster/meta_executor.go

```diff
@@ -0,0 +1,157 @@
+package cluster
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/influxdata/influxdb"
+	"github.com/influxdata/influxdb/influxql"
+	"github.com/influxdata/influxdb/services/meta"
+)
+
+// MetaExecutor executes meta queries on all data nodes.
+type MetaExecutor struct {
+	mu             sync.RWMutex
+	timeout        time.Duration
+	pool           *clientPool
+	maxConnections int
+	Logger         *log.Logger
+	Node           *influxdb.Node
+
+	MetaClient interface {
+		DataNode(id uint64) (ni *meta.NodeInfo, err error)
+		DataNodes() ([]meta.NodeInfo, error)
+	}
+}
+
+// NewMetaExecutor returns a new initialized *MetaExecutor.
+func NewMetaExecutor() *MetaExecutor {
+	return &MetaExecutor{
+		timeout:        DefaultWriteTimeout,
+		pool:           newClientPool(),
+		maxConnections: 1000,
```

Comment:
Is this max connections per node?  If so, default is too high.  We use 3 per-node for writes.  Also, should be a constant like: https://github.com/influxdata/influxdb/blob/master/cluster/config.go#L21